internal/config/config.go

```diff
@@ -98,3 +98,43 @@ func PostgresContainerImage(cluster *v1beta1.PostgresCluster) string {
 func PGONamespace() string {
 	return os.Getenv("PGO_NAMESPACE")
 }
+
+// VerifyImageValues checks that all container images required by the
+// spec are defined. If any are undefined, a list is returned in an error.
+func VerifyImageValues(cluster *v1beta1.PostgresCluster) error {
+
+	var images []string
+
+	if PGBackRestContainerImage(cluster) == "" {
+		images = append(images, "crunchy-pgbackrest")
+	}
+	if PGAdminContainerImage(cluster) == "" &&
+		cluster.Spec.UserInterface != nil &&
+		cluster.Spec.UserInterface.PGAdmin != nil {
+		images = append(images, "crunchy-pgadmin4")
+	}
+	if PGBouncerContainerImage(cluster) == "" &&
+		cluster.Spec.Proxy != nil &&
+		cluster.Spec.Proxy.PGBouncer != nil {
+		images = append(images, "crunchy-pgbouncer")
+	}
+	if PGExporterContainerImage(cluster) == "" &&
+		cluster.Spec.Monitoring != nil &&
+		cluster.Spec.Monitoring.PGMonitor != nil &&
+		cluster.Spec.Monitoring.PGMonitor.Exporter != nil {
+		images = append(images, "crunchy-postgres-exporter")
```

Comment:
⛏️ Not a blocker, but I wonder if the order of checks here should have the `*Image` func last? Like: is this cluster asking for a proxy? Do we have an image for that requested proxy?

Reply:
Most of the time, I believe evaluation will stop at the first check the way it's written, so I figure that's worth something. I could see it either way I think.